### PR TITLE
Feature/chests deployment

### PR DESCRIPTION
**Pull request by kermankohli**

### CODE REVIEW

File: contracts/gods-unchained/deployment/02_season_one.ts [hunk 1]
```diff
@@ -258,7 +297,59 @@ export class SeasonOneStage implements DeploymentStage {
...
+    legendaryPack: string,
+    referral: string,
+    escrow: string,
+    processor: string,
+  ) {
+    console.log('** Deploying Legendary Chest **');
+
+    const unsignedTx = await Chest.getDeployTransaction(
+      this.wallet,
+      'GU:S1: Rare Chest',
```
Comment (alonnovy) [inline contracts/gods-unchained/deployment/02_season_one.ts:338]: Should this say "Legendary Chest"?

File: contracts/gods-unchained/deployment/02_season_one.ts [hunk 2]
```diff
@@ -153,20 +174,29 @@ export class SeasonOneStage implements DeploymentStage {
...
-    return contract.address;
+    const unsignedTx = await S1Sale.getDeployTransaction(this.wallet);
+    unsignedTx.nonce = await this.wallet.getTransactionCount();
+    const signedTx = await this.wallet.sendTransaction(unsignedTx);
+    const receipt = await signedTx.wait();
+    return receipt.contractAddress;
   }
 
   async deployReferral(): Promise<string> {
     console.log('** Deploying Referral **');
```
Comment (alex-connolly) [inline contracts/gods-unchained/deployment/02_season_one.ts:194]: Can this stuff be handled with Isaac directly? Is it the nonce issue?

File: contracts/gods-unchained/scripts/MAIN_CONTRACTS.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+module.exports = [
+  { GU_Forwarder: 'Forwarder' },
+  { GU_Cards: 'Cards' },
+  { GU_OpenMinter: 'OpenMinter' },
+  { GU_Fusing: 'Fusing' },
+  { GU_PromoFactory: 'PromoFactory' },
+  { GU_S1_Vendor: 'S1Vendor' },
+  { GU_S1_Raffle: 'Raffle' },
+  { GU_S1_Sale: 'S1Sale' },
+  { GU_S1_Referral: 'Referral' },
```
Comment (alex-connolly) [inline contracts/gods-unchained/scripts/MAIN_CONTRACTS.js:17]: Need to add chests to these.

File: contracts/platform/scripts/MAIN_CONTRACTS.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+ * These contracts are primary contracts that will:
+ *      - Have ABIs exported to @imtbl/artifacts
+ *      - Verified on Etherscan
+ *
+ */
+
+module.exports = [
+  { IM_Beacon: 'Beacon' },
+  { IM_Processor: 'Processor' },
+  { IM_TestVendor: 'TestVendor' },
```
Comment (alex-connolly) [inline contracts/platform/scripts/MAIN_CONTRACTS.js:11]: Don't need to export TestVendor

File: packages/addresses/src/outputs.ts [hunk 5]
```diff
@@ -57,24 +77,23 @@ export const outputs = {
...
+      GU_Fusing: '0x6DfFF22588BE9b3ef8cf0aD6Dc9B84796F9fB45f',
+      GU_PromoFactory: '0xcFC18CEc799fBD1793B5C43E773C98D4d61Cc2dB',
+      GU_S1_Vendor: '0x10aDd991dE718a69DeC2117cB6aA28098836511B',
+      GU_S1_Raffle: '0x8d61158a366019aC78Db4149D75FfF9DdA51160D',
+      GU_S1_Sale: '0x131855DDa0AaFF096F6854854C55A4deBF61077a',
+      GU_S1_Referral: '0xB69e673309512a9D726F87304C6984054f87a93b',
+      GU_S1_Epic_Pack: '0xE86bB98fcF9BFf3512C74589B78Fb168200CC546',
+      GU_S1_Rare_Pack: '0xDc688D29394a3f1E6f1E5100862776691afAf3d2',
+      GU_S1_Shiny_Pack: '0xb7C9b454221E26880Eb9C3101B3295cA7D8279EF',
+      GU_S1_Legendary_Pack: '0x6000EcA38b8B5Bba64986182Fe2a69c57f6b5414',
```
Comment (alonnovy) [inline packages/addresses/src/outputs.ts:96]: Will it output the Chest addresses also?

Comment (kermankohli) [reply packages/addresses/src/outputs.ts:96]: Yup

File: contracts/gods-unchained/deployment/02_season_one.ts (hunk 1 above)
Comment (kermankohli) [reply contracts/gods-unchained/deployment/02_season_one.ts:338]: Good catch, fixed.

File: contracts/gods-unchained/deployment/02_season_one.ts (hunk 2 above)
Comment (kermankohli) [reply contracts/gods-unchained/deployment/02_season_one.ts:194]: Yeah, this can be added into Isaac directly

**Pull request merged**